lstchain/scripts/lstchain_dl1_to_dl2.py

```diff
@@ -112,7 +104,34 @@ def main():
     
     #Apply the models to the data
 
-    dl2 = dl1_to_dl2.apply_models(data, cls_gh, reg_energy, reg_disp_vector, custom_config=config)
+    #Source-independent analysis
+    if not config['source_dependent']:
+        data = filter_events(data,
+                             filters=config["events_filters"],
+                             finite_params=config['regression_features'] + config['classification_features'],
+                         )
+
+        dl2 = dl1_to_dl2.apply_models(data, cls_gh, reg_energy, reg_disp_vector, custom_config=config)
+
+    #Source-dependent analysis
+    if config['source_dependent']:
+        dl1_srcdep_keys = get_dataset_srcdep_keys(args.input_file)
+        dl2_srcdep_list = []
+        dl2_srcdep_keys_list = []
+
+        for i, k in enumerate(dl1_srcdep_keys):
+            data_srcdep = pd.read_hdf(args.input_file, key=k)
+            data_with_srcdep_param = pd.concat([data, data_srcdep], axis=1)
+            data_with_srcdep_param = filter_events(data_with_srcdep_param,
```

Comment:
Do you want to filter based on source dependent parameter as well?
It looks pretty inefficient to apply the same filters every time again, maybe split them into src independent and source dependent filters somehow? That would also avoid the `i==0` part below

Reply:
In principle I assume source-independent parameters are used for the filtering, but source-dependent parameters could also be filtered because they are needed to apply random forest and removed if they are infinite.
Moreover if the parameters which we want to apply filtering are not in the data, an error is shown now. It means the same filtering cannot be used for each source-dep/indep parameters.
I'd like to consider more, any comments are welcome.